GitUI/CommandsDialogs/FormPush.cs

```diff
@@ -504,13 +504,14 @@ private bool HandlePushOnExit(ref bool isError, FormProcess form)
                 {
                     if (!form.ProcessArguments.Contains(" -f ") && !form.ProcessArguments.Contains(" --force"))
                     {
-                        if (GitCommandHelpers.VersionInUse.SupportPushForceWithLease)
+                        if(form.ProcessArguments.StartsWith("push ", StringComparison.OrdinalIgnoreCase))
```

Comment:
>I personally prefer this over having the "safe code" with the else asserting false, but perhaps we should ask @jbialobr to weigh in on this.

First of all, I want to thank you for reviewing PRs. This is a community project and I don't want to force my coding preferences. Let's wait for @azarkevich to see his opinion (pros and cons). I will have to decide only If you won't find an agreement.